recipes/compilers/meta.yaml

```diff
@@ -0,0 +1,204 @@
+{% set commands = {
+  'binutils': {
+    'linux': ['ar', 'as', 'ld', 'nm', 'size', 'strings', 'strip',
+              'addr2line', 'c++filt', 'dwp', 'elfedit', 'gprof', 'ld.bfd',
+              'ld.gold', 'objcopy', 'objdump', 'ranlib', 'readelf'],
+    'osx': ['ar', 'as', 'ld', 'nm', 'size', 'strings', 'strip',
+            'checksyms', 'codesign_allocate', 'indr', 'install_name_tool',
+            'libtool', 'lipo', 'nmedit', 'otool', 'pagestuff', 'ranlib',
+            'redo_prebinding', 'seg_addr_table', 'seg_hack', 'segedit'],
+  },
+  'c-compiler': {
+    'linux': ['cc', 'ct-ng.config', 'cpp', 'gcc', 'gcc-ar', 'gcc-nm',
+              'gcc-ranlib', 'gcov', 'gcov-dump', 'gcov-tool'],
+    'osx': ['clang'],
```

Comment:
Shouldn't this be keyed based on what the results of `compiler('c')` are rather than the arch? That or we shouldn't use `compiler('c')` in the run deps and instead specify the compilers explicitly